ie_serving/server/predict_utils.py

```diff
@@ -61,32 +62,46 @@ def prepare_input_data(models, model_name, version, data, rest):
                 message = str(e)
                 logger.debug("PREDICT prepare_input_data make_ndarray error: "
                              "{}".format(message))
-                return True, message, None, code
+                return True, message, code
         else:
             tensor_input = np.asarray(data[requested_input_blob])
+        # Validate shape if shape not in auto mode
+        if models[model_name].engines[version].shape_info.mode != ShapeMode.AUTO:

```

Comment:
Ehhh I think it's time to pass to `prepare_input_data` only three variables :inference engine object, data, service_type.

After that change readable of our code will increase :)